readthedocs/projects/tasks/search.py

```diff
@@ -40,112 +42,215 @@ def fileify(version_pk, commit, build, search_ranking, search_ignore):
+):
     """
-    Create imported files for version.
+    Create imported files and search index for the build of the version.
+
+    If the version is external, we don't create a search index for it, only imported files.
+    After the process is completed, we delete the files and search index that
+    don't belong to the current build id.
 
-    :param version: Version instance
-    :param commit: Commit that updated path
-    :param build: Build id
+    :param search_index: If provided, it will be used as the search index name,
+     otherwise the default one will be used.
     """
-    # Re-create all objects from the new build of the version
     storage_path = version.project.get_storage_path(
-        type_='html', version_slug=version.slug, include_file=False
+        type_="html", version_slug=version.slug, include_file=False
     )
+    html_files_to_index = []
+    html_files_to_save = []
+    reverse_rankings = list(reversed(search_ranking.items()))
     for root, __, filenames in build_media_storage.walk(storage_path):
         for filename in filenames:
             # We don't care about non-HTML files
-            if not filename.endswith('.html'):
+            if not filename.endswith(".html"):
                 continue
 
             full_path = build_media_storage.join(root, filename)
 
             # Generate a relative path for storage similar to os.path.relpath
-            relpath = full_path.replace(storage_path, '', 1).lstrip('/')
+            relpath = full_path.replace(storage_path, "", 1).lstrip("/")
+
+            skip_search_index = False
+            if version.is_external:
+                # Never index files from external versions.
+                skip_search_index = True
+            else:
+                for pattern in search_ignore:
+                    if fnmatch(relpath, pattern):
+                        skip_search_index = True
+                        break
 
             page_rank = 0
-            # Last pattern to match takes precedence
-            # XXX: see if we can implement another type of precedence,
-            # like the longest pattern.
-            reverse_rankings = reversed(list(search_ranking.items()))
-            for pattern, rank in reverse_rankings:
-                if fnmatch(relpath, pattern):
-                    page_rank = rank
-                    break
-
-            ignore = False
-            for pattern in search_ignore:
-                if fnmatch(relpath, pattern):
-                    ignore = True
-                    break
-
-            # Create imported files from new build
-            HTMLFile.objects.create(
+            # If the file is ignored, we don't need to check for its ranking.
+            if not skip_search_index:
+                # Last pattern to match takes precedence
+                # XXX: see if we can implement another type of precedence,
+                # like the longest pattern.
```

Comment:
```suggestion
                # Last pattern to match takes precedence
                # TODO: see if we can implement another type of precedence,
                # like the longest pattern.
```

Reply:
I think we can just remove this note, don't think we will be changing how our search priority works, that will be a breaking change. I like how the current precedence work.